pkg/healthsrv/healthz_handler.go

```diff
@@ -0,0 +1,73 @@
+package healthsrv
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	s3 "github.com/aws/aws-sdk-go/service/s3"
+	"github.com/deis/builder/pkg/gitreceive/log"
+	"github.com/deis/builder/pkg/sshd"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+type healthZRespBucket struct {
+	Name         string    `json:"name"`
+	CreationDate time.Time `json:"creation_date"`
+}
+
+func convertBucket(b *s3.Bucket) healthZRespBucket {
+	return healthZRespBucket{
+		Name:         *b.Name,
+		CreationDate: *b.CreationDate,
+	}
+}
+
+type healthZResp struct {
+	Namespaces       []string            `json:"k8s_namespaces"`
+	S3Buckets        []healthZRespBucket `json:"s3_buckets"`
+	SSHServerStarted bool                `json:"ssh_server_started"`
+}
+
+func healthZHandler(nsLister NamespaceLister, bLister BucketLister, serverCircuit *sshd.Circuit) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// There's a race between the boolean eval and the HTTP error returned (the server could start up between the two), but k8s will repeat the health probe request and effectively re-evaluate the boolean. The result is that the server may not start until the next probe in those cases
+		if serverCircuit.State() != sshd.ClosedState {
+			str := fmt.Sprintf("SSH Server is not yet started")
+			log.Err(str)
+			http.Error(w, str, http.StatusServiceUnavailable)
+			return
+		}
+		lbOut, err := bLister.ListBuckets(&s3.ListBucketsInput{})
+		if err != nil {
+			str := fmt.Sprintf("Error listing buckets (%s)", err)
+			log.Err(str)
+			http.Error(w, str, http.StatusServiceUnavailable)
+			return
+		}
+		var rsp healthZResp
+		for _, buck := range lbOut.Buckets {
+			rsp.S3Buckets = append(rsp.S3Buckets, convertBucket(buck))
+		}
+
+		nsList, err := nsLister.List(labels.Everything(), fields.Everything())
```

Comment:
Why not `framework.NewInformer` like [here](https://github.com/kubernetes/contrib/blob/master/service-loadbalancer/service_loadbalancer.go#L569) to not hit the apiserver unnecessarily